homeassistant/helpers/__init__.py

```diff
@@ -24,6 +26,8 @@ def config_per_platform(config: ConfigType, domain: str) -> Iterable[tuple[Any,
         if not isinstance(platform_config, list):
             platform_config = [platform_config]
 
+        item: ConfigType
+        platform: str | None
```

Comment:
I wasn't sure about ` | None` here.
If there is an `AttributeError`, should it still yield?

Reply:
I think so... not 100% sure. I believe automation and script use this as well (and ignore platform, as that is irrelevant).

However, this can be mitigated in the other places like mailbox by doing:

```py
[
  ...
  for p_type, p_config in config_per_platform(config, DOMAIN)
  if p_type is not None
]
```

Reply:
Should I move this to a preliminary PR?

Reply:
I've created #63890 - I think it makes sense to split it out.